docs/guides/federation.md

```diff
@@ -2,6 +2,183 @@
 title: Federation
 ---
 
-# Federation
+# Apollo Federation Guide
 
-Documentation coming soon
+Strawberry supports [Apollo Federation][1] out of the box, that means that you
+can create services using Strawberry and federate them via Apollo Gateway.
+
+> _NOTE_: we don’t have a gateway server, you’d need to always use the Apollo
+> Gateway for this.
+
+Apollo Federation allows you to combine multiple GraphQL APIs into one. This can
+be extremely useful when working with a service oriented architecture.
+
+## Federated schema example
+
+Let’s look at an example on how to implement Apollo Federation using Strawberry.
+Let's assume we have an application with two services that both expose a GraphQL
+API:
+
+1. `book`: a service to manage all the books we have
+2. `reviews`: a service to manage book reviews
+
+### Books service
+
+Our `book` service might look something like this:
+
+```python
+@strawberry.federation.type(keys=["id"])
+class Book:
+    id: strawberry.ID
+    title: str
+
+def get_all_books() -> List[Book]:
+    return [Book(id=1, title="The Dark Tower")]
+
+@strawberry.type
+class Query:
+    all_books: List[Book] = strawberry.field(resolver=get_all_books)
+
+schema = strawberry.federation.Schema(query=Query)
+```
+
+We defined two types: `Book` and `Query`, where `Query` has only one field that
+allows us to fetch all the books.
+
+Notice that the `Book` type used the `strawberry.federation.type` decorator, as
+opposed to the normal `strawberry.type`, this new decorator extends the base one
+and allows us to define federation specific attributes on the type.
+
+In this case we are telling federation that the key to uniquely identify a book
+is the `id` field.
+
+> Federation keys can be thought primary keys. They are used by the gateway to
+> query types between multiple services and then join them into the augmented
+> type.
+
+### Reviews service
+
+Let’s look at how our review service looks like: we want to define a type for a
+review but also extend the book to have a list of reviews.
+
+```python
+@strawberry.type
+class Review:
+    body: str
+
+def get_reviews(book: "Book") -> List[Review]:
+    return [
+      Review(body=f"This is review number {index} for {book.id}")
+      for index in range(book.reviews_count)
+    ]
+
+@strawberry.federation.type(extend=True, keys=["id"])
+class Book:
+    id: strawberry.ID = strawberry.federation.field(external=True)
+    reviews_count: int
+    reviews: List[Review] = strawberry.field(resolver=get_reviews)
+
+    @classmethod
+    def resolve_reference(cls, id: strawberry.ID):
+        return Book(id, reviews_count=3)
+```
+
+Now things are looking more interesting, the `Review` type is a GraphQL type
+that holds the content of the review.
+
+Meanwhile we are able to extend the `Book` type by using
+`strawberry.federation.type` again and passing `extend=True` as a parameter.
+This tells federation that we are extending an already existing type.
```

Comment:
```suggestion
We've also been able to extend the `Book` type by using again
`strawberry.federation.type`, this time passing `extend=True` as an argument.
This is important because we need to tell federation that we are extending a
type that already exists, not creating a new one.
```